pkg/storage/iterators.go

```diff
@@ -397,3 +399,124 @@ func NewConditionsFilteredTupleKeyIterator(iter TupleKeyIterator, filter TupleKe
 func IterIsDoneOrCancelled(err error) bool {
 	return errors.Is(err, ErrIteratorDone) || errors.Is(err, context.Canceled) || errors.Is(err, context.DeadlineExceeded)
 }
+
+type OrderedCombinedIterator struct {
+	mu      *sync.Mutex
+	once    *sync.Once
+	pending []TupleIterator // GUARDED_BY(mu)
+	mapper  TupleMapper
+}
+
+var _ TupleIterator = (*OrderedCombinedIterator)(nil)
+
+type TupleMapper func(t *openfgav1.Tuple) string
+
+func UserMapper() TupleMapper {
+	return func(t *openfgav1.Tuple) string {
+		return t.GetKey().GetUser()
+	}
+}
+
+func ObjectMapper() TupleMapper {
+	return func(t *openfgav1.Tuple) string {
+		return t.GetKey().GetObject()
+	}
+}
```

Comment:
TODO: these belong, conceptually, in `internal/graph/tuplemapper.go`, but i don't want to have `storage` import `graph`.